Model/Order/Invoice/Total/Surcharge.php

```diff
@@ -0,0 +1,42 @@
+<?php
+/**
+ * @by SwiftOtter, Inc. 4/11/17
+ * @website https://swiftotter.com
+ **/
+
+namespace SwiftOtter\ShippingSurcharge\Model\Order\Invoice\Total;
+
+use SwiftOtter\ShippingSurcharge\Model\Surcharge as SurchargeModel;
+
+class Surcharge extends \Magento\Sales\Model\Order\Invoice\Total\AbstractTotal
+{
+    public function collect(\Magento\Sales\Model\Order\Invoice $invoice)
+    {
+        $surcharge = 0;
+        $baseSurcharge = 0;
+
+        /** @var \Magento\Sales\Model\Order\Invoice\Item $item */
+        foreach ($invoice->getAllItems() as $item) {
+            $orderItem = $item->getOrderItem();
+            $orderItemQty = $orderItem->getQtyOrdered();
+
+            if (!$orderItemQty || $orderItem->isDummy() || $item->getQty() < 1) {
```

Comment:
The negation operator is a tricky one that I'm learning to try to avoid. It rarely seems to enhance the readability of the code, since it can mean the value can be one of at least three things: `false`, `null`, or 0. If you're checking for a specific one, I'd suggest making that more clear.